mars/deploy/oscar/ray.py

```diff
@@ -82,31 +89,46 @@ def __init__(self,
 
     async def start(self):
         address_to_resources = dict()
-        supervisor_node_address = f'ray://{self._cluster_name}/0'
-        address_to_resources[supervisor_node_address] = {
+        supervisor_exclusive_node = self._config \
+            .get('cluster', {}) \
+            .get('ray', {}) \
+            .get('supervisor_exclusive_node', DEFAULT_SUPERVISOR_EXCLUSIVE_NODE)
+        self.supervisor_address = process_placement_to_address(self._cluster_name, 0, 0)
+        address_to_resources[node_placement_to_address(self._cluster_name, 0)] = {
             'CPU': 1,
             # 'memory': self._supervisor_mem
         }
-        worker_node_addresses = []
-        for worker_index in range(1, self._worker_num + 1):
-            worker_node_address = f'ray://{self._cluster_name}/{worker_index}'
-            worker_node_addresses.append(worker_node_address)
-            address_to_resources[worker_node_address] = {
-                'CPU': self._worker_cpu,
-                # 'memory': self._worker_mem
-            }
+        worker_addresses = []
+        if supervisor_exclusive_node:
+            for worker_index in range(1, self._worker_num + 1):
+                worker_address = process_placement_to_address(self._cluster_name, worker_index, 0)
+                worker_addresses.append(worker_address)
+                worker_node_address = node_placement_to_address(self._cluster_name, worker_index)
+                address_to_resources[worker_node_address] = {
+                    'CPU': self._worker_cpu,
+                    # 'memory': self._worker_mem
+                }
+        else:
+            for worker_index in range(self._worker_num):
+                worker_address = process_placement_to_address(self._cluster_name, worker_index, int(worker_index == 0))
```

Comment:
supervisor can have some sub pools. The worker address should start after the last supervisor sub pool address instead of `1`. For example, if the node parallelism is 4 and the supervisor pool address is `ray:://cluster/0/0, ray:://cluster/0/1`, the worker address should be `ray:://cluster/0/2, ray:://cluster/0/3`

Reply:
Can we add a test with supervisor parallelism of `2` to cover this case?

Reply:
The `create_supervisor_actor_pool` is called with a fixed n_process=0. There is no sub pool of supervisor actor pool. @qinxuye Will the supervisor actor pool have a sub pool in the future?

Reply:
Yes, for local it's 0, but for a distributed setting, it must be >= 0

Reply:
> Yes, for local it's 0, but for a distributed setting, it must be >= 0

Thanks. @chaokunyang I will make the n_process configurable for ray.